pkg/agent/stats/collector.go

```diff
@@ -108,6 +113,23 @@ func (m *Collector) collect() *statsCollection {
 	acnpStatsMap := map[types.UID]map[string]*statsv1alpha1.TrafficStats{}
 	anpStatsMap := map[types.UID]map[string]*statsv1alpha1.TrafficStats{}
 
+	if m.multicastEnabled {
+		multicastStatsMap := m.multicastQuerier.GetIGMPReportNPStats()
+		for networkPolicyType, uidStatsMap := range multicastStatsMap {
+			for uid, ruleStats := range uidStatsMap {
+				for ruleName, ruleStats := range ruleStats {
+					switch networkPolicyType {
+					case cpv1beta.K8sNetworkPolicy:
+						addPolicyStatsUp(npStatsMap, ruleStats, uid)
+					case cpv1beta.AntreaClusterNetworkPolicy:
+						addRuleStatsUp(acnpStatsMap, ruleStats, uid, ruleName)
+					case cpv1beta.AntreaNetworkPolicy:
```

Comment:
Need to think about if ANP is needed for Multicast. In my opinion, ACNP should work for Multicast, since it is complecated if we check namespace on the Pod receivers with ANP rules?

Reply:
ANP is supported for multicast NP and multicast NP stats. Please check this thread https://github.com/antrea-io/antrea/pull/3660#discussion_r876633681